src/client/Microsoft.Identity.Client/ApiConfig/AbstractPublicClientAcquireTokenParameterBuilder.cs

```diff
@@ -21,6 +26,47 @@ internal AbstractPublicClientAcquireTokenParameterBuilder(IPublicClientApplicati
             PublicClientApplicationExecutor = publicClientApplicationExecutor;
         }
 
+#if DESKTOP
+        /// <summary>
+        ///  Modifies the token acquisition request so that the acquired token is a Proof of Possession token (PoP), rather than a Bearer token. 
+        ///  PoP tokens are similar to Bearer tokens, but are bound to the HTTP request and to a cryptographic key, which MSAL can manage on Windows.
+        ///  See https://aka.ms/msal-net-pop
+        /// </summary>
+        /// <param name="httpRequestMessage">An HTTP request to the protected resource which requires a PoP token. The PoP token will be cryptographically bound to the request.</param>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item>This is an experimental API. The method signature may change in the future without involving a major version upgrade.</item>
+        /// <item> Add the PoP token in an Authorization header, just like a bearer token. See <seealso cref="AuthenticationResult.CreateAuthorizationHeader"/> for details.</item>
+        /// <item> The PoP token is bound to the HTTP request, more specifically to the HTTP method (GET, POST, etc.) and to the Uri (path and query, but not query parameters). </item>
+        /// <item> MSAL creates, reads and stores a key securely on behalf of the Windows user. </item>
+        /// </list>
+        /// </remarks>
+        public T WithPoPAuthenticationScheme(HttpRequestMessage httpRequestMessage) 
```

Comment:
@jmprieur @henrik-me - Please review the public API. There are 2 possibilities for the param `HttpRequestMessage` (which embeds uri, method etc.) and `uri` + `method` explicitly. 

I like `HttpRequestMessge` more because in the future it is possible that we will bind not only the Uri and the Method, but also the forms params and headers, which users could "turn on", similar to how Wilson will validate the PoP tokens.

Also, we may consider adding the `Authentication: PoP ey...` header directly to the HttpRequestMessage.

Reply:
I agree @bgavrilMS that we should use the abstractions available in .NET, and this has potential to evolve. We'll just need to document what members are used today. 

Adding the authentication to the HttpRequestMessage (in a re-entrant way), would also be good. We'd also probably be able to provide an end to end with the HttpRequestMessage (calling the resource, getting a 403 and an www.Authenticate header, parsing it, and finding that it needs Pop?